packages/mural-client/src/fetch.ts

```diff
@@ -269,27 +272,35 @@ export const authorizeHandler = (config: TokenHandlerConfig) => async (
  * @param opts.store Whether the tokens should automatically be persisted in the provided
  * storage. This is pretty handy unless you want to manage the tokens on your own.
  *
+ * @param opts.redirectUri Set a custom redirectUri as query parameter redirect_uri
+ * to be handled by your application's redirect uri handler.
+ *
  * @returns The token pair issued from the MURAL OAuth service
  */
-export const requestTokenHandler = (config: TokenHandlerConfig) => async (
-  code: string,
-  state: string,
-  opts = { store: false },
-): Promise<Session> => {
-  // validate that the state hasn't been tampered
-  if (!validateState(state)) throw new Error('INVALID_STATE');
-
-  const url = `${config.requestTokenUri}?code=${code}`;
-  const session = await fetch(url, { method: 'GET' })
-    .then(checkStatus)
-    .then(res => res.json());
-
-  if (opts.store) {
-    fetchConfig.sessionStore.set(session);
-  }
+export const requestTokenHandler =
+  (config: TokenHandlerConfig) =>
+  async (
+    code: string,
+    state: string,
+    opts: RequestTokenHandlerOptions = { store: false },
+  ): Promise<Session> => {
+    // validate that the state hasn't been tampered
+    if (!validateState(state)) throw new Error('INVALID_STATE');
+
+    const url = `${config.requestTokenUri}?code=${code}${
```

Comment:
Could not use URL here because for some usages, we just pass in a path.

Reply:
```suggestion
  const params = new URLSearchParams();
  params.set('code', code);
  if (opts.redirectUri) params.set('redirect_uri', opts.redirectUri);

  const url = `${config.requestTokenUri}?${params.toString()}`;
```

**Nitpick**: Here's my suggestion, to me this is easier to read

Reply:
Done! Thanks!

Reply:
Plus, tested locally and 👍